### DISCUSSION TITLE
Is plain-vanilla Javascript `adopt`?

### DISCUSSION THREAD

**Discussion by balanza** [pagopa/technology-radar#29] [General]
Javascript is everywhere, so of course we're gonna use it eventually. But can we state we `adopt` Javascript?

As an engineering community, should we have a position about the use of plain vanilla Javascript in our applications?

As per my experience in PagoPA I would say:
* Javascript is `hold` for application development, meaning you should prefer Typescript
* (maybe) Javascript is `adopt` for scripting, meaning an application may require some tooling for which the type fatigue of Typescript is overkill

**Comment by gunzip** [comment]
Keep in mind that for a technology to be placed in the 'adopt' quadrant, it must have already been implemented by various projects across different teams, demonstrating its applicability and relevance to the entire organization's needs.

**Reply by balanza** [reply]
>  it must have already been implemented

This is necessary but not sufficient, right?

**Reply by balanza** [reply]
The point is: 
should we consider "ok" an application written in plain JavaScript? No.
Should consider "ok" a CLI tool written in plain JavaScript? Why not

**Reply by ZorbaDimatteo** [reply]
For example, in SEND we use js when developing AWS lambda to avoid unnecessary transpiling tasks and time. I'm for `adopt` for scripting, small serveless "disposable" applications, or even for small common library

**Reply by lucacavallaro** [reply]
a CLI tool can be complex (think about our client generator), and a web app can be extremely simple (like `io-link`)

maybe JS may only be recommended for small one file scripts or configuration files

**Reply by balanza** [reply]
> unnecessary transpiling tasks

@ZorbaDimatteo do you think we can better define what "unnecessary" means in our context? I would like to put it in a positive perspective.

_We use javascript when..._

**Reply by ZorbaDimatteo** [reply]
Ok, it could be something like:

_We want to use JavaScript when crafting small scripts or functions for services like AWS Lambda, as it provides a hassle-free development experience that allows us to concentrate more on business logic. JavaScript's dynamic typing and lightweight syntax eliminate the need for setting up a complex environment, enabling a quick and straightforward development process. This simplicity is particularly beneficial for small-scale applications, helping us avoid unnecessary configuration and streamline the focus on the essential business logic. Moreover, JavaScript's extensive ecosystem ensures easy integration and access to a wealth of resources, further enhancing the efficiency of serverless development on platforms like AWS Lambda._

**Reply by balanza** [reply]
Thanks, it makes sense.

It sounds to me like a tactical choice rather than a community best practice. Just like unit tests:
* Should we write unit tests for every feature? Yes
* Do I write unit tests for every feature? No, sometimes I skip unit testing for trivial functions
* Should we advise it's better not to do unit tests for trivial functions? No
